packages/theme/src/cli/services/push.ts

```diff
@@ -34,13 +34,21 @@ export async function push(theme: Theme, session: AdminSession, options: PushOpt
   const themeChecksums = await fetchChecksums(theme.id, session)
   const themeFileSystem = await mountThemeFileSystem(options.path)
 
-  const results = await uploadTheme(theme, session, themeChecksums, themeFileSystem, options)
+  const {uploadResults, renderThemeSyncProgress} = await uploadTheme(
```

Comment:
This method is hanging / not terminating when I 🎩 btw!

Reply:
I think [these changes](https://github.com/Shopify/cli/pull/4367/files#diff-d857e2b583dcee2197fc538eb4d35aabb5ce83cee18ebb970ece7854dda8ba07) will resolve that issue for us